securedrop/requirements/update_python_dependencies

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+
+# Usage: ./update-python-dependencies
+# Run periodically to keep Python application, test, and development
+# requirements up-to-date.
+
+set -eC
+
+WORKING_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+declare -r WORKING_DIR
+declare -r VENV='review_env'
+declare -ra REQUIREMENTS_FILE_PREFIXES=('securedrop' 'test' 'develop')
+
+# This script should not be run with an active virtualenv. Calling deactivate
+# does not work reliably, so instead we quit with an error message.
+if [[ -n "${VIRTUAL_ENV}" ]]; then
+  printf '\nERROR: Please deactivate your virtualenv then re-run this script.\n'
+  exit 1
+fi
+
+# If pip and virtualenv are not available, try to install them with apt-get. If
+# apt-get is not available, quit with an error message.
+if ! command -v pip virtualenv > /dev/null; then
+  if command -v sudo apt-get > /dev/null; then
+    sudo apt-get -qq install python-virtualenv python-pip
+  else
+    printf '\nERROR: Virtualenv and pip must be installed to run this script.\n'
+    exit 1
+  fi
+fi
+
+pushd "${WORKING_DIR}"
+# Create a temporary virtualenv for the SecureDrop Python packages in our
+# requirements directory and make sure it's cleaned up on exit.
+trap 'rm -rf "${VENV}" && popd' EXIT
+
+virtualenv -p python2 "${VENV}"
+# shellcheck disable=SC1090
+source "${VENV}"/bin/activate
+pip install -U pip-tools
```

Comment:
This provides a simple, reliable, and relatively fast way to ensure we have the latest `pip` and `pip-tools`. Further, it doesn't make any modifications to user-created virtual environments nor does it make changes that may move any of our virtual machines further away from our production configuration. Having the latest versions installed is important because (i) it will allow us to do the most with #1617 and (ii) it might avoid problems between older version of these packages and PyPi or the options we specify to pip in this script, among other potential incompatibilities.

Basically, the time expenditure is very little (an extra few seconds) and repeated runs of the script are not realistic in practice.